pkg/client/unversioned/request.go

```diff
@@ -585,8 +592,20 @@ func (r *Request) Watch() (watch.Interface, error) {
 	if client == nil {
 		client = http.DefaultClient
 	}
+	if r.backoffMgr == nil {
+		panic("nil backoffMgr should never occur")
```

Comment:
I think in this case, the code is going to be read hopefully far more than any nil pointer backtrace from line 598. It's a tradeoff, but here I'd prefer to keep the code slightly cleaner.